centreon/www/front_src/src/ApiTokens/TokenListing/Actions/Activate/Activate.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState } from 'react';
+
+import { useQueryClient } from '@tanstack/react-query';
+
+import { Method, useMutationQuery } from '@centreon/ui';
+import { Switch } from '@centreon/ui/components';
+
+import { patchTokenEndpoint } from '../../../api/endpoints';
+import { labelActiveOrRevoked } from '../../../translatedLabels';
+import { Row } from '../../models';
+
+const Activate = ({ row }: Row): React.JSX.Element => {
+  const queryClient = useQueryClient();
+
+  const [isRevoked, setIsRevoked] = useState<boolean>(row?.isRevoked);
+
+  const { mutateAsync } = useMutationQuery({
+    getEndpoint: () =>
+      patchTokenEndpoint({ tokenName: row?.name, userId: row?.user.id }),
+    method: Method.PATCH,
+    onError: () => setIsRevoked(!isRevoked),
```

Comment:
in the case of error is the component does not rerender ?  for my point of view for both cases onError/onSuccess the component should rerender and the isRevoked would be initialized to  raw?.isRevoked

Reply:
In case of error the component reverts to a previous state

Reply:
even if you put the isRevoked to true , on error case the raw.isRevoked is always false , since you initialize the isRevoked to raw?.isRevoked so it will be true , no need to revok a rerender using setIsRevoke(!isRevoked)

Reply:
my bad , the initialition will happen when the component destroyed